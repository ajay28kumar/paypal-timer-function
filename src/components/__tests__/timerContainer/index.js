import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { renderHook, act } from "@testing-library/react-hooks";

import TimerContainer, { useTimerDetails } from "../../timerContainer";

describe("timer component tests", () => {
  it("snapshot testing", () => {
    const app = shallow(<TimerContainer />);
    expect(toJson(app)).toMatchSnapshot();
  });
  it("AddTimer is rendered", () => {
    const app = shallow(<TimerContainer />);
    expect(app.find("AddTimer").exists()).toBe(true);
  });

  it("expect Timer component is not rendered initially", () => {
    const app = shallow(<TimerContainer />);
    expect(app.find("Timer").exists()).toBeFalsy();
  });
  it("test hook function", () => {
    const { result } = renderHook(() => useTimerDetails(0));
    act(() => {
      result.current.setTimerDetails([{ id: "261", startCounter: 90 }]);
    });
    expect(result.current.timerDetails.length).toBe(1);
  });
});
