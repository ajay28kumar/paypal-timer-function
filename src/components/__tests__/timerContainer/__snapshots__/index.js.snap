// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`timer component tests snapshot testing 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-container-1"
  container={true}
  spacing={4}
>
  <AddTimer
    setTimerDetails={[Function]}
    timerDetails={Array []}
  />
</WithStyles(ForwardRef(Grid))>
`;
